main/src/mill/main/AndroidHiltModule.scala

```diff
@@ -0,0 +1,23 @@
+package mill.main
+
+import coursier.LocalRepositories
+import coursier.core.Repository
+import coursier.maven.MavenRepository
+import mill.api.PathRef
+import mill.define.{Discover, ExternalModule, Target}
+import mill.util.MillModuleUtil.millProjectModule
+
+object AndroidHiltModule extends ExternalModule {
+  def repositories: Seq[Repository] = Seq(
+    LocalRepositories.ivy2Local,
+    MavenRepository("https://repo1.maven.org/maven2"),
+    MavenRepository("https://maven.google.com/")
+  )
+
+  lazy val millDiscover = Discover[this.type]
+
+  def toolsClasspath: Target[Seq[PathRef]] = Target {
+    millProjectModule("mill-main-android-hilt", repositories)
```

Comment:
This should give you a deprecation warning. 

Instead use `Dep.millProjectModule` to get the dependency representing the module and use `jvmWorker().defaultResolver().classpath` to resolve it as usual.